src/NuGet.Core/NuGet.Protocol.Core.Types/UserAgent.cs

```diff
@@ -2,81 +2,30 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Globalization;
 using System.Net.Http;
-using NuGet.Versioning;
 
 namespace NuGet.Protocol.Core.Types
 {
     public static class UserAgent
     {
-        private static NuGetVersion _NuGetClientVersion;
-        private static string _OSVersion;
-
-        private const string UserAgentTemplate = "{0}/{1} ({2})";
-        private const string UserAgentWithHostTemplate = "{0}/{1} ({2}, {3})";
-
-        public const string NuGetClientName = "NuGet Client V3";
-
-        /// <summary>
-        /// To be set by NuGet Clients such as NuGet Extension for Visual Studio or nuget.exe
-        /// </summary>
-        public static string UserAgentString = string.Empty;
-
-        /// <summary>
-        /// Create user agent string with template of "{0}/{1} ({2})", where {0} is client name,
-        /// {1} is NuGetClientVersion and {2} is OSVersion. {1} and {2} are automatically computed.
-        /// </summary>
-        /// <param name="client">Client name</param>
-        /// <returns></returns>
-        public static string CreateUserAgentString(string client)
+        static UserAgent()
         {
-            if (client == null)
-            {
-                throw new ArgumentNullException(nameof(client));
-            }
-
-            if (NuGetTestMode.Enabled)
-            {
-                client = NuGetTestMode.NuGetTestClientName;
-            }
-
-            return string.Format(
-                CultureInfo.InvariantCulture,
-                UserAgentTemplate,
-                client,
-                NuGetClientVersion,
-                OSVersion);
+            // Set default user agent string
+            UserAgentString = new UserAgentStringBuilder().Build();
```

Comment:
@emgarten This used to be initialized to `string.Empty` by default. This change may in fact also fix https://github.com/NuGet/Home/issues/2047?